x-pack/plugins/task_manager/server/saved_objects/migrations.ts

```diff
@@ -146,3 +150,22 @@ function pipeMigrations(...migrations: TaskInstanceMigration[]): TaskInstanceMig
   return (doc: SavedObjectUnsanitizedDoc<TaskInstanceWithDeprecatedFields>) =>
     migrations.reduce((migratedDoc, nextMigration) => nextMigration(migratedDoc), doc);
 }
+
+function resetAttemptsAndStatusForTheTasksWithoutSchedule(
+  doc: SavedObjectUnsanitizedDoc<TaskInstanceWithDeprecatedFields>
+): SavedObjectUnsanitizedDoc<TaskInstanceWithDeprecatedFields> {
+  if (doc.attributes.taskType.startsWith('alerting:')) {
+    if (!doc.attributes.schedule?.interval) {
```

Comment:
We should look specifically for tasks with status `failed`. I don't think we want to set `running` tasks back to `idle`.